app/models/report.rb

```diff
@@ -482,4 +482,128 @@ def self.report_trending_search(report)
+    GROUP BY uv.user_id
+    SQL
+
+    flag_count_query = <<~SQL
+    SELECT pa.agreed_by_id AS user_id,
+    COUNT(*) AS flag_count
+    FROM post_actions pa
+    WHERE pa.agreed_by_id = ANY(ARRAY#{mod_ids})
+    AND pa.post_action_type_id = ANY(ARRAY#{PostActionType.flag_types_without_custom.values})
+    GROUP BY pa.agreed_by_id
+    SQL
+
+    topic_count_query = <<~SQL
+    SELECT t.user_id AS user_id,
+    COUNT(*) AS topic_count
+    FROM topics t
+    WHERE t.user_id = ANY(ARRAY#{mod_ids})
+    GROUP BY t.user_id
+    SQL
+
+    post_count_query = <<~SQL
+    SELECT p.user_id AS user_id,
+    COUNT(*) AS post_count
+    FROM posts p
+    WHERE p.user_id = ANY(ARRAY#{mod_ids})
+    GROUP BY p.user_id
+    SQL
+
+    DB.query(time_read_query).each do |row|
+      mod_data[row.user_id][:time_read] = row.time_read
+    end
+
+    DB.query(flag_count_query).each do |row|
+      mod_data[row.user_id][:flag_count] = row.flag_count
+    end
+
+    DB.query(topic_count_query).each do |row|
+      mod_data[row.user_id][:topic_count] = row.topic_count
+    end
+
+    DB.query(post_count_query).each do |row|
+      mod_data[row.user_id][:post_count] = row.post_count
+    end
+
+    mod_data.each do |k, v|
+      report.data << v
+    end
+  end
+
+  def self.report_recent_flags(report)
+    report.data = []
+    flag_types = PostActionType.flag_types_without_custom
+
+    flag_actions = PostAction.joins("INNER JOIN posts ON posts.id = post_actions.post_id")
+                     .where(post_action_type_id: flag_types.values)
+                     .where("post_actions.created_at >= ? AND post_actions.created_at <= ?", report.start_date, report.end_date)
+                     .select('post_actions.post_action_type_id,
+                              posts.id,posts.topic_id,
+                              posts.user_id AS poster_id,
+                              post_actions.created_at,
+                              post_actions.agreed_at,
+                              post_actions.disagreed_at,
+                              post_actions.deferred_at,
+                              post_actions.user_id,
+                              post_actions.agreed_by_id,
+                              post_actions.disagreed_by_id,
+                              post_actions.deferred_by_id')
+
+    flag_actions.each do |action|
+      flag = {}
+      flag[:action_type] = flag_types.key(action.post_action_type_id).to_s
+      flag[:post_id] = action.id
+      flag[:topic_id] = action.topic_id
+      if action.agreed_at
+        flag[:response_time] = action.agreed_at - action.created_at
+      elsif action.disagreed_at
+        flag[:response_time] = action.disagreed_at - action.created_at
+      elsif action.deferred_at
+        flag[:response_time] = action.deferred_at - action.created_at
+      else
+        flag[:response_time] = nil
+      end
+      flag[:poster_id] = action.poster_id
+      flag[:poster_username] = User.find(action.poster_id).username
```

Comment:
This will generate N+1 requests. Which will become slower as the number of flag actions grows. To prevent this, you need to move that query in your first query ;)

Reply:
Thanks! I assumed that was wrong. Will fix.